### PR TITLE
Add Allow and Restrict to PluginConfigOptionValue

### PR DESCRIPTION
**Pull request by echlebek**
This commit allows Allow and Restrict fields to PluginConfigOptionValue.
The purpose of these fields is to facilitate a uniform method of simple
option validation through equality testing. The equality testing is
provided via go-cmp. The zero values of Allow and Restrict result in the
same behaviour as before they were added, so this change is
backwards-compatible.

Signed-off-by: Eric Chlebek <eric@sensu.io>

Note to reviewers: this change was based on another change still in a branch, so it will be rebased on master after the "refactorin" PR is merged.

### CODE REVIEW

File: sensu/plugin.go [hunk 1]
```diff
@@ -289,20 +492,126 @@ func validateEvent(event *corev2.Event) error {
...
-	case *[]string:
-		if err := json.Unmarshal([]byte(valueStr), value); err == nil {
-			return nil
+func (p *PluginConfigOption[T]) SetValue(valueStr string) (err error) {
+	if p.Value == nil {
+		return errors.New("PluginConfigOption.Value not set!")
+	}
+	defer func() {
+		if err == nil {
+			err = p.validateAllowRestrict()
```
Comment (c-kruse) [inline sensu/plugin.go:501]: It looks like validateAllowRestrict is only getting called for annotation overrides and not when the config is coming from viper.

Comment (echlebek) [reply sensu/plugin.go:501]: :facepalm: DOH

Comment (c-kruse) [reply sensu/plugin.go:501]: Maybe we need to add a validation loop in the workflow functions? That's about the best suggestion I could think up.

Comment (echlebek) [reply sensu/plugin.go:501]: Oh, perhaps not.

**Review by c-kruse — APPROVED** [review]
LGTM!

**Pull request merged**